views/recipe.ejs

```diff
@@ -0,0 +1,80 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title><%= recipe ? "Edit Recipe" : "Add Recipe" %></title>
+  <!-- Bootstrap CSS -->
+  <link href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css" rel="stylesheet">
+  <style>
+    /* Custom CSS for form layout */
+    .ingredient {
+      margin-bottom: 10px;
+    }
+  </style>
+</head>
+<body>
+  <div class="container mt-5">
+    <h1 class="mb-4"><%= recipe ? "Edit Recipe" : "Add Recipe" %></h1>
+    <form action="/recipes<%= recipe ? '/' + recipe._id + '?_method=PUT' : '' %>" method="POST">
+      <input type="hidden" name="_csrf" value="<%= csrfToken %>">
+      
+      <div class="form-group">
+        <label for="recipeName">Recipe Name:</label>
+        <input type="text" class="form-control" id="recipeName" name="recipeName" value="<%= recipe ? recipe.recipeName : '' %>" required>
+      </div>
+      
+      <div class="form-group">
+        <label for="instructions">Instructions:</label>
+        <textarea class="form-control" id="instructions" name="instructions" rows="4" required><%= recipe ? recipe.instructions : '' %></textarea>
+      </div>
+      
+      <div id="ingredients">
+        <% if (recipe && recipe.ingredients.length > 0) { %>
+          <% recipe.ingredients.forEach((ingredient, index) => { %>
+            <div class="ingredient">
```

Comment:
Nitpick:
Duplicate code for this `div class=“ingredient”` element and for the `button` to remove the ingredient. Your if/else statement should be moved inside and within these two elements.